hw2/TommyLin/q2/test_vector.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+import pytest
+import math
+import numpy as np
+import _vector
+
+def angle(vec1, vec2):
+    dot_product = np.dot(vec1, vec2)
+    len1 = np.dot(vec1, vec1)
+    len2 = np.dot(vec2, vec2)
+    return np.arccos(dot_product / (np.sqrt(len1 * len2)))
+
+def testInvalidVector():
+    with pytest.raises(TypeError):
```

Comment:
good test

Reply:
I will put something other than a number as arguments

Reply:
Using string as arguments made "TypeError" also. There are more input test ran, only array with two element is allowed. Any other conditions as follow throw "TypeError".
        _vector.angle([0, 0], [])
        _vector.angle([], [0, 0])
        _vector.angle([], [])
        _vector.angle([1], [1,1])
        _vector.angle(["1"], [1,1])
TypeError check  is accomplished by compiler and runtime system not in my source. This makes source code clear without checking "Type Error". These testings help me to know compiler will do type check strictly. But for testing, maybe one condition is enough, because I know runtime shall test type strictly.
BTW. There is one more issue with my test. Any one of these condition throw "TypeError". It should test separately to make sure each of condition throw "TypeError".
    with pytest.raises(TypeError):
        _vector.angle([0, 0], [])
    with pytest.raises(TypeError):
        _vector.angle([], [0, 0])